message-tags-3.3.md

```diff
@@ -0,0 +1,71 @@
+---
+title: IRCv3.3 Message Tags
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Kiyoshi Aman"
+    period: 2016
+    email: "kiyoshi.aman@gmail.com"
+  -
+    name: "James Wheare"
+    period: 2016
+    email: "james@irccloud.com"
+---
+## Introduction
+
+This specification adds a new capability for general message tags support and
+a prefix for expressing client-to-client tags.
+
+## Motivation
+
+A new capability is required to allow negotiating support for client-to-client
+tags, as well as any well-formed, unrecognised tag. This removes the need to
+negotiate a capability for each new tag-only specification, and frees servers
+from having to filter tags individually when responding to clients.
+
+## Architecture
+
+### Capabilities
+
+This specification adds the `message-tags` capability. Clients which request
+this capability indicate that they are capable of parsing all tags, regardless
+of support for tags specified in other IRCv3.2 or later specifications.
+
+### Tags
+
+Client-to-client tags are client-initiated tags that servers MUST attach as-is
+to any relevant event relayed to other clients. A client-to-client tag is prefixed
+with a plus sign (`+`) and otherwise conforms to the format specified in
+[IRCv3.2 tags][irctags].
+
+Client-to-client tags MUST be relayed on `PRIVMSG` and `NOTICE` events, and
+MAY be forwarded on other events.
+
+The expected client behaviour of individual client-to-client tags SHOULD be defined
+in separate specifications, in the same way as server-initiated tags.
```

Comment:
I'd love to see some clarification in the spec if tag keys themselves within a message are unique or can be used multiple times.

Whether the following is valid/expected or not:

```
@+icon=https://example.com/favicon.png;+icon=something :url_bot!bot@example.com PRIVMSG #channel :Example.com: A News Story
```